test/extractor/ConfigurationExtractor.test.ts

```diff
@@ -59,6 +59,41 @@ describe('ConfigurationExtractor', () => {
       expect(loggerSpy.getAllMessages('warn')).toHaveLength(0);
     });
 
+    it('should be able to handle a specified FHIR version', () => {
+      const ig = JSON.parse(
+        fs.readFileSync(path.join(__dirname, 'fixtures', 'simple-ig.json'), 'utf-8')
+      );
+      const result = ConfigurationExtractor.process([ig, ...resources], '4.5.0');
+      expect(result.config.fhirVersion).toEqual(['4.5.0']);
+      expect(loggerSpy.getAllMessages('warn')).toHaveLength(0);
+    });
+
+    it('should be able to handle a specified FHIR version when FHIR version is missing', () => {
+      const ig = JSON.parse(
+        fs.readFileSync(path.join(__dirname, 'fixtures', 'missing-fhir-version-ig.json'), 'utf-8')
+      );
+      const result = ConfigurationExtractor.process([ig, ...resources], '5.0.0');
+      expect(result.config.fhirVersion).toEqual(['5.0.0']);
+      expect(loggerSpy.getAllMessages('warn')).toHaveLength(2);
+      expect(loggerSpy.getMessageAtIndex(-2, 'warn')).toMatch(
+        'ImplementationGuide missing properties needed to generate configuration file: fhirVersion. These properties will be inferred based on FHIR definitions.'
+      );
+      expect(loggerSpy.getLastMessage('warn')).toMatch(
+        'FHIR Version mismatch error: specified version is: 5.0.0 while detected version is: 4.0.1. GoFSH will use the specified version (5.0.0), but this may result in additional processing errors.'
+      );
+    });
+
+    it('should log an error when the specified FHIR version is different from detected FHIR version', () => {
```

Comment:
This is also a really small suggestion to make the test description more accurate now that it is a warning.

```suggestion
    it('should log a warning when the specified FHIR version is different from detected FHIR version', () => {
```